src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,749 @@
+         * @param textToAdd The data to be written.
+         * @throws IOException If the file does not exist.
+         */
+        private void writeToFile(String filePath, String textToAdd) throws IOException {
+            FileWriter fw = new FileWriter(filePath);
+            fw.write(textToAdd + "\n");
+            fw.close();
+        }
+
+        /**
+         * Appends the task to a file.
+         *
+         * @param filePath The storage file to be written to.
+         * @param textToAdd The data to be written.
+         * @throws IOException If the file does not exist.
+         */
+        private void appendToFile(String filePath, String textToAdd) throws IOException {
+            FileWriter fw = new FileWriter(filePath, true);
+            fw.write(textToAdd + "\n");
+            fw.close();
+        }
+
+        /**
+         * Saves the given TaskList to a file.
+         *
+         * @param l The TaskList to be saved.
+         */
+        public void saveAsFile(TaskList l) {
+            boolean isFirst = true;
+            if (l.isEmpty()) {
+                try {
+                    writeToFile(filePath, " ");
+                } catch (IOException e) {
+                    System.out.println("Something went wrong uwu: " + e.getMessage());
+                }
+            }
+            for (Task t : l.getList()) {
+                try {
+                    if (isFirst) {
+                        writeToFile(filePath, t.saveToData());
+                        isFirst = false;
+                    } else {
+                        appendToFile(filePath, t.saveToData());
+                    }
+                } catch (IOException e) {
+                    System.out.println("Something went wrong uwu: " + e.getMessage());
+                }
+            }
+        }
+
+    }
+
+    public static class TaskList {
+
+        private List<Task> leest;
+
+        public TaskList() {
+            this.leest = new ArrayList<>();
+        }
+
+        public void add(Task t) {
+            this.leest.add(t);
+        }
+
+        public boolean isEmpty() {
+            return this.leest.isEmpty();
+        }
+
+        public int size() {
+            return this.leest.size();
+        }
+
+        public Task get(int i) {
+            return this.leest.get(i);
+        }
+
+        public void remove(Task t) {
+            this.leest.remove(t);
+        }
+
+        public List<Task> getList() {
+            return this.leest;
+        }
+
+    }
+
+    public class Parser {
+
+        /**
+         * Parses a given input and returns Strings as necessary based on the input.
+         * The main driver of the Duke class.
+         *
+         * @param leest The current TaskList.
+         * @param input Input that is being read in.
+         */
+        public String parse(TaskList leest, String input) {
+
+            String[] split = input.split("\\s+");
+
+            switch (split[0]) {
+            case "list":
```

Comment:
I like your use of switch case statements! However, do note that we may need to add the "//Fallthrough" under each case where a break  is supposed to be placed, but unreachable!